### PR TITLE
Naev 0.9.0 has Been Released

### PR DESCRIPTION
**Pull request by ProjectSynchro**
Howdy!
I'm an upstream developer for Naev and noticed this Flatpak is out of date so I updated it to the best of my ability (I am fairly new to Flatpak).

Basic Overview of Changes:
- [x] Removed old screenshots
- [x] Updated patch for metainfo
- [x] Ported manifest from JSON to YAML (Sorry, I am far more comfortable with YAML so I figured I'd convert the JSON with [this](https://github.com/flatpak/flatpak-builder-tools/blob/master/flatpak-json2yaml.py) and go from there)
- [x] Added new dependencies for Naev 0.9.0

I'm not fully sure how the workflow goes for updating flatpaks let alone what FlatHub expects in terms of setup.

Also: if you would be so kind as to add me as a contributor on here that would be fantastic (See: https://github.com/orgs/naev/people)

Thanks! and Happy Holidays

### CODE REVIEW

**Comment by lionirdeadman** [comment]
Just wanna give you a warm welcome, glad that someone upstream has stepped up to maintain!

I'd like to note that you may want to upgrade the runtime to 21.08 as this is the latest runtime version. (20.08 is going EOL in ~8-9 months so it's not an issue if you wish to do this later)

**Comment by ProjectSynchro** [comment]
> Just wanna give you a warm welcome, glad that someone upstream has stepped up to maintain!
> 
> I'd like to note that you may want to upgrade the runtime to 21.08 as this is the latest runtime version. (20.08 is going EOL in ~8-9 months so it's not an issue if you wish to do this later)

Glad to be here :smile: 

What is the typical policy on using the Runtime? With AppImages the policy is build on the oldest versions of everything, should Flatpaks always track the newest thing?

File: org.naev.Naev.yaml [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+runtime-version: '21.08'
+sdk: org.freedesktop.Sdk
+command: naev
+rename-icon: naev
+rename-desktop-file: org.naev.naev.desktop
+rename-appdata-file: org.naev.naev.metainfo.xml
+finish-args:
+  - --socket=pulseaudio
+  - --socket=wayland
+  - --socket=system-bus
```
Comment (barthalion) [inline org.naev.Naev.yaml:12]: What is system-bus needed for here?

File: org.naev.Naev.yaml [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+  - --socket=x11
```
Comment (barthalion) [inline org.naev.Naev.yaml:13]:
```suggestion
  - --socket=fallback-x11
```

File: org.naev.Naev.yaml [hunk 3]
```diff
@@ -0,0 +1,124 @@
+id: org.naev.Naev
+runtime: org.freedesktop.Platform
+runtime-version: '21.08'
+sdk: org.freedesktop.Sdk
+command: naev
+rename-icon: naev
+rename-desktop-file: org.naev.naev.desktop
+rename-appdata-file: org.naev.naev.metainfo.xml
```
Comment (barthalion) [inline org.naev.Naev.yaml:8]: FYI, ideally, these renames should happen upstream.

**Comment by barthalion** [comment]
> What is the typical policy on using the Runtime? With AppImages the policy is build on the oldest versions of everything, should Flatpaks always track the newest thing?

freedesktop-sdk runtimes are maintained for 2 years, so it's generally a good idea to upgrade to the latest one sooner than later.

File: metainfo.patch [hunk 4]
```diff
@@ -0,0 +1,82 @@
+--- a/org.naev.naev.metainfo.xml	2021-12-22 23:53:07.924279000 -0500
++++ b/org.naev.naev.metainfo.xml	2021-12-24 04:41:29.462095392 -0500
+@@ -1,11 +1,10 @@
+ <?xml version="1.0" encoding="UTF-8"?>
+ <component type="desktop-application">
+-  <id>org.naev.naev</id>
++  <id>org.naev.naev.desktop</id>
```
Comment (hfiguiere) [inline metainfo.patch:7]:
```suggestion
+  <id>org.naev.naev</id>
```

it was correct before

**Review by hfiguiere — CHANGES_REQUESTED** [review]

**Comment by hfiguiere** [comment]
> What is the typical policy on using the Runtime? With AppImages the policy is build on the oldest versions of everything, should Flatpaks always track the newest thing?

The difference is AppImage ship with everything. Flatpak apps don't include the runtime. The runtime is shared with other apps, and freedesktop-sdk is maintained for 2 years. So ideally the flatpak should be updated to use the latest runtime whenever possible.

File: org.naev.Naev.yaml (hunk 1 above)
Comment (ProjectSynchro) [reply org.naev.Naev.yaml:12]: After reading some more of the documentation with some coffee in me it's definitely not needed here.

File: org.naev.Naev.yaml (hunk 2 above)
Comment (ProjectSynchro) [reply org.naev.Naev.yaml:13]: Wasn't aware of this option, makes the most sense to me!

File: org.naev.Naev.yaml (hunk 3 above)
Comment (ProjectSynchro) [reply org.naev.Naev.yaml:8]: Gotcha,
I'll make the changes upstream for the names, that should make it into a later point release.

File: metainfo.patch (hunk 4 above)
Comment (ProjectSynchro) [reply metainfo.patch:7]: Good to know. I'll update the patch.

**Comment by ProjectSynchro** [comment]
> > What is the typical policy on using the Runtime? With AppImages the policy is build on the oldest versions of everything, should Flatpaks always track the newest thing?
> 
> The difference is AppImage ship with everything. Flatpak apps don't include the runtime. The runtime is shared with other apps, and freedesktop-sdk is maintained for 2 years. So ideally the flatpak should be updated to use the latest runtime whenever possible.

Makes sense, In AppImage land you want to target the oldest glibc version for the widest compatibility, but if that isn't an issue with Flatpak I see no problems staying proactive about keeping as up-to-date as possible.

File: org.naev.Naev.yaml [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+finish-args:
+  - --socket=pulseaudio
+  - --socket=wayland
+  - --socket=fallback-x11
+  - --device=dri
```
Comment (hfiguiere) [inline org.naev.Naev.yaml:13]: `- --share=ipc` is still required for fallback-x11. It cause a slower execution path otherwise (if running X11).

Comment (ProjectSynchro) [reply org.naev.Naev.yaml:13]: Good to know!

**Review by barthalion — APPROVED** [review]

**Comment by barthalion** [comment]
LGTM. I've just sent you an invitation to collaborators – so you should be able to merge this PR yourself.

**Comment by ProjectSynchro** [comment]
> LGTM. I've just sent you an invitation to collaborators – so you should be able to merge this PR yourself.

Cool beans, Thanks!

Happy Christmas Eve :smile:

**Pull request merged**